engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ReadArgumentCheckNode.java

```diff
@@ -479,14 +477,11 @@ private static final class LazyCheckRootNode extends RootNode {
     @Child private ReadArgumentCheckNode check;
 
     static final FunctionSchema SCHEMA =
-        new FunctionSchema(
-            FunctionSchema.CallerFrameAccess.NONE,
-            new ArgumentDefinition[] {
-              new ArgumentDefinition(0, "delegate", null, null, ExecutionMode.EXECUTE)
-            },
-            new boolean[] {true},
-            new CallArgumentInfo[0],
-            new Annotation[0]);
+        FunctionSchema.newBuilder()
```

Comment:
`FunctionSchema` builder! Nice.

Reply:
Indeed. There were so many Java constructors that I just had to refactor it to a proper *builder* pattern.